pkg/nodeconfig/nodeconfig.go

```diff
@@ -521,7 +521,15 @@ func (nc *nodeConfig) Deconfigure() error {
 	if err != nil {
 		return err
 	}
-	if err := nc.Windows.Deconfigure(nc.wmcoNamespace, wicdKC); err != nil {
+	if err := nc.Windows.RunWICDCleanup(nc.wmcoNamespace, wicdKC); err != nil {
+		return fmt.Errorf("unable to cleanup the Windows instance: %w", err)
+	}
+
+	// Wait for reboot annotation removal. This prevents deleting the node until the node no longer needs reboot.
+	if err := metadata.WaitForRebootAnnotationRemoval(context.TODO(), nc.client, nc.node.Name); err != nil {
+		return err
+	}
+	if err := nc.Windows.Deconfigure(); err != nil {
 		return fmt.Errorf("error deconfiguring instance: %w", err)
 	}
```

Comment:
Theres an ordering requirement thats not clear here.
If its solely between `RunWICDCleanup` and `WaitForRebootAnnotationRemoval` it would be best to move them to a separate documented function.

Reply:
At this point even `nc.generateWICDKubeconfig()` needs to move into the separate documented function.

Reply:
Makes sense, thanks for pointing this out